app/routes/users.js

```diff
@@ -4,22 +4,50 @@ const userService = require('../services').userService;
 
 /* GET users listing. */
 router.get('/', function(req, res, next) {
-  res.send('respond with a resource');
+  return req.sessionID ? res.redirect('/#!/signin') : res.redirect('/#!/signup');
 });
 
-router.post('/', function(req, res, next) {
+router.post('/', function(req, res) {
   userService.create(req.body.username, req.body.password, req.body.email)
     .then((user) => {
       delete user.password;
 
       // Store the user in the session.
       req.session.user = user;
       res.json(user);
-    }).catch((err) => {
+    })
+    .catch((err) => {
       res.status(500).json({
         error: 'error creating user'
       });
     });
 });
 
+router.post('/signin', function(req, res) {
+  userService.signin(req.body)
+    .then((user) => {
+      delete user.password;
+      req.session.user = user;
+      res.json(user);
+    })
+    .catch((err) => {
+      res.status(500).json({
```

Comment:
Also, if you want this `catch` to be triggered then something in the auth will have to `reject` or throw an error.